### ISSUE TITLE
Strange behaviour in dsl

### ISSUE THREAD

**Issue by voanhduy1512** [caskroom/homebrew-cask#2615]
@phinze, I may need your advise here. When doing #2532, i tried to change following lines in dsl.rb

```ruby
ARTIFACT_TYPES.each do |type|
      define_method(type) do |*args|
        artifacts[type] << args
      end
    end
```
to 
```ruby
ARTIFACT_TYPES.each do |type|
      define_method(type) do |args|
        artifacts[type] << args
      end
    end
end
```
Because each artifact will take only one argument, so with current `*args`. This code
```ruby
class Abc < Cask
...
link 'Abc.app';
...
end
```
will parse args into an Array `['Abc.app']`. Then with `artifacts[type] << args`, `artifacts[type]` will become `[['Abc.app']]`.
It add one more [] then `args` alone (not *), and i feel it is bad. So i change `*args` to `args` and ran a quick test with modified alfred cask like this 

```ruby
class Alfred < Cask
  url 'http://cachefly.alfredapp.com/Alfred_2.1.1_227.zip'
  homepage 'http://www.alfredapp.com/'
  version '2.1.1_227'
  sha1 'db28d2a5c655a4611c780c3f46252530118ddd9d'
  link 'Alfred 2.app', :target => 'AnotherApp.app'
  link 'Alfred 2.app/Contents/Preferences/Alfred Preferences.app'
end
````
and it install just fine.
Then i run all the test, in fact @rolandwalker #2543 and it gave me error on 
```
/Users/anhduyvo/Projects/homebrew-cask/lib/cask/dsl.rb:66:in `block (2 levels) in <module:ClassMethods>': wrong number of arguments (2 for 1) (ArgumentError)
```
Exactly the line i've just modified
```ruby
define_method(type) do |args|
``` 

So may i ask why i can run and install success on a real cask but it give me an error on test. I tried to investigate but no success

**Comment by vitorgalvao** [comment]
Is this still an issue?

**Comment by voanhduy1512** [comment]
I will closed it as #2532 is merged and i forgot  about the issue now :D
